src/components/learning-guide/practice-by-type-button.cjsx

```diff
@@ -16,13 +15,25 @@ module.exports = React.createClass
 
   onClick: ->
     {courseId} = @props
-    all = LearningGuide.Student.store.getSortedSections(@props.courseId)
-    sections = if @props.practiceType is 'weaker' then _.first(all, 4) else _.last(all, 4)
-    page_ids = _.chain(sections).pluck('page_ids').flatten().uniq().value()
-    @context.router.transitionTo('viewPractice', {courseId}, {page_ids})
+    page_ids = LearningGuide.Student.store.getPracticePages(@props.courseId, @props.practiceType)
+    unless _.isEmpty(page_ids)
```

Comment:
I'm not sure that's a reasonable thing to do here.  It seems to me that the page_ids could be empty for non bug related reasons.  For instance, right now on tutor-demo, student01 has only once assignment that is past due and has not been started.  The performance report endpoint is returning an empty set, as it should because there's not any data to report yet.  

Your suggestion about the button's not rendering anything when empty is good, but I'd suggest that check move up into the parent component.  Otherwise we'll end up with a bold "**Practice**" heading and no buttons below it.  The button could still have the check as it does here, just in case the parent doesn't perform the check .